### PR TITLE
Leagues Functionality

### PR DESCRIPTION
**Pull request by mikemaddem**

### CODE REVIEW

File: project-templates/staff/leagues/league_create.html [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        <label>Prize 1</label>
+         form.prize1 }}
+        <br>
+
+        <label>Prize 2</label>
+         form.prize2 }}
+        <br>
+
+        <label>Prize 3</label>
+         form.prize3 }}
```
Comment (techlover1) [inline project-templates/staff/leagues/league_create.html:129]: Missing opening {{ on prizes

File: project-templates/staff/leagues/league_create.html [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        <br>
+
+        <label>Active</label>
+        {{ form.active }}
+        <br>
+
+        <label>Ruleset</label>
+        {{ form.ruleset }}
```
Comment (techlover1) [inline project-templates/staff/leagues/league_create.html:137]: Maybe this should be part of league settings?

File: project-templates/staff/leagues/league_detail.html [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+{% load static %}
+
+{% block title %}
+    League #{{ league.id }}
+{% endblock %}
+
+{% block body %}
+
+
+    <p><a href="{% url 'staff:list_league' %}"><i class="fa fa-arrow-left"></i> <strong>Back to Tournament
```
Comment (techlover1) [inline project-templates/staff/leagues/league_detail.html:11]: Should be back to league list

File: project-templates/staff/leagues/league_divisions_list.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                <th>League Name</th>
+                </thead>
+            </tr>
+        <tbody>
+            {% for division in divisions %}
+                <tr scope="">
+                    <td scope="row"><a href="{% url 'staff:detail_division' divid=division.id pk=league.id %}">{{ division.id }}</a></td>
+                    <td><a
+                            href="{% url 'staff:division_matches_league' pk=league.id divid=division.pk %}">
+                        <btn class="btn btn-primary">Matches#{{ division.id }}</btn>
```
Comment (techlover1) [inline project-templates/staff/leagues/league_divisions_list.html:25]: This should probably be changed to "Matches for division {{division.id}}"

File: project-templates/staff/leagues/league_division_addteam.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        {% endfor %}
+    {% endif %}
+    <h1><a href="{% url 'staff:list_league' %}">Return to league list</a></h1>
+
+    <form enctype="multipart/form-data" method="post">
+        {% csrf_token %}
+        {{ field.errors }}
+
+        <label>Team id</label>
+        {{ form.teamid }}
```
Comment (techlover1) [inline project-templates/staff/leagues/league_division_addteam.html:38]: Make this more user friendly

File: project-templates/staff/leagues/league_addmatch.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    <form enctype="multipart/form-data" method="post">
+        {% csrf_token %}
+        {{ field.errors }}
+
+        <label>Away team id</label>
+        {{ form.awayteam }}
+        <br>
+
+        <label>Home team id</label>
+        {{ form.hometeam }}
```
Comment (techlover1) [inline project-templates/staff/leagues/league_addmatch.html:42]: Make these more user friendly

File: project-templates/staff/leagues/league_division_detail.html [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        <td><a href="{% url 'staff:team_detail' pk=match.awayteam.pk %}">{{ match.awayteam.name }}</a></td>
+        <td><a href="{% url 'staff:team_detail' pk=match.hometeam.pk %}">{{ match.hometeam.name }}</a></td>
+
+    </tr>
+    {% endfor %}
+
+    </table>
+    <h1>Teams</h1>
+    {% for team in division.teams.all %}
+    {{ team.name }}
```
Comment (techlover1) [inline project-templates/staff/leagues/league_division_detail.html:60]: This doesn't actually work

File: leagues/views.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+def detail_league_teams(request, pk):
+    pass
+
+
+def detail_league_divisions(request, pk):
+    pass
+
+
+def detail_league_rules(request, pk):
+    pass
```
Comment (techlover1) [inline leagues/views.py:28]: these functions actually need to do something

File: project-templates/staff/leagues/league_division_addteam.html (hunk 5 above)
Comment (mikemaddem) [reply project-templates/staff/leagues/league_division_addteam.html:38]: how? in a later issue potentially, I don't see anything wrong with this.

File: project-templates/leagues/leagues_list.html [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+                            {% if leagues.count == 0 %}
+                                <b>No Leagues yet!</b>
+                            {% endif %}
+                            {% for league in leagues %}
+                                <div class="col-xs-12 col-sm-6 col-md-4 league">
+                                    <img src="
+                                            {% if ESPORTS_MODE %}
+
+                                        {% if league.game.image != "" %}{{ MEDIA_URL }}{{ league.game.image }}{% else %}{% static 'binge/images/logo.png' %}{% endif %}"
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/leagues/leagues_list.html:52]: removed

File: project-templates/leagues/leagues_list.html [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+                                        ><br>
+                                    {% else %}
+                                        {{ league.sport.name }}
+                                    {% endif %}
+                                    <h6><b>{{ league.name }}</b></h6>
+                                    <br/>
+                                    <ul>
+                                        {% if ESPORTS_MODE %}
+                                            <li><b>Platform:</b> {{ league.platform.name }}</li>{% endif %}
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/leagues/leagues_list.html:61]: fixed

File: project-templates/leagues/leagues_list.html [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+                                        <li><b>Starts:</b> {{ league.start }}</li>
+                                        <li><b>Teams: </b> {{ league.teams.count }}/{{ league.size }}</li>
+                                        {% if ESPORTS_MODE %}
+                                            <li><b>Game:</b> {{ league.game.name }}</li>{% endif %}
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/leagues/leagues_list.html:65]: fixed

File: project-templates/leagues/leagues_list.html [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+                                        {% elif league.teamformat == 5 %}
+                                            6v6
+                                        {% endif %}</b></li>
+                                    </ul>
+                                    <ul>
+
+                                        <li><b>Registration Opens: (Local time)</b>
+                                            <script>
+                                                const date = new Date("{{ league.open_register|safe }}");
+                                                console.log(date);
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/leagues/leagues_list.html:85]: removed

File: project-templates/leagues/leagues_list.html [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+                                                document.write(date.toLocaleString());
+                                            </script>
+                                        </li>
+                                        <li><b>Registration Closes (Local time)</b>
+                                            <script>
+                                                const date2 = new Date("{{ league.close_register|safe }}");
+                                                console.log(date2);
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/leagues/leagues_list.html:92]: removed

File: project-templates/staff/leagues/league_create.html [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+        <br>
+        {% endif %}
+
+        <div class="form-group">
+            <div class="input-group">
+                <label>Start Time</label>
+                {{ form.start }}
+            </div>
+        </div>
+        <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css" rel="stylesheet">
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/staff/leagues/league_create.html:106]: moved

File: project-templates/staff/leagues/league_create.html [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+        {% endfor %}
+        {% for error in form.non_field_errors %}
+            <div class="alert alert-danger">
+                <strong>{{ error|escape }}</strong>
+            </div>
+        {% endfor %}
+    {% endif %}
+    <h1><a href="{% url 'staff:list_league' %}">Return to league list</a></h1>
+    <br>
+    <a href="{% url 'staff:tournamentrulesetcreate' %}"><i class="fa fa-plus"></i> Create a new ruleset</a>
```
[the remark this answers is not in the document]
Comment (mikemaddem) [reply project-templates/staff/leagues/league_create.html:33]: added league settings

File: project-templates/staff/leagues/league_create.html (hunk 1 above)
Comment (mikemaddem) [reply project-templates/staff/leagues/league_create.html:129]: had to readd prizes, fixed

File: project-templates/staff/leagues/league_create.html (hunk 2 above)
Comment (mikemaddem) [reply project-templates/staff/leagues/league_create.html:137]: I don't think so, settings are more for a set game and adding this quick field has minimal effect.

File: project-templates/staff/leagues/league_detail.html (hunk 3 above)
Comment (mikemaddem) [reply project-templates/staff/leagues/league_detail.html:11]: changed text

File: project-templates/staff/leagues/league_division_detail.html (hunk 7 above)
Comment (mikemaddem) [reply project-templates/staff/leagues/league_division_detail.html:60]: fixed

**Comment by mikemaddem** [comment]
500 error is because you need to pass the league pk in the url....

**Comment by mikemaddem** [comment]
> Templates missing content:
> 
> * leagues/league_detail.html
> * leagues/league_rules.html
> * leagues/league_teams.html
> * staff/leagues/league_settings_detail.html
> ~~* staff/leagues/league_teams.html~~
> * staff/leagues/league_teams_add.html

fixed 500 error

**Comment by mikemaddem** [comment]
@all-contributors add @techlover1 for reviewed pull requests

**Comment by mikemaddem** [comment]
@all-contributors add @mikemaddem for reviewed pull requests

**Comment by mikemaddem** [comment]
@all-contributors add @techlover1 for reviewed pull requests

**Comment by mikemaddem** [comment]
@all-contributors add @techlover1 for reviewed pull requests

**Comment by mikemaddem** [comment]
@all-contributors add @techlover1 for pull requests

**Comment by mikemaddem** [comment]
@all-contributors add @mikemaddem for projectManagement

**Comment by mikemaddem** [comment]
@all-contributors add @techlover1 for review

**Pull request merged**